targets/lenovo-x1-carbon.nix

```diff
@@ -0,0 +1,181 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Generic x86_64 computer -target
+{
+  self,
+  lib,
+  nixos-generators,
+  nixos-hardware,
+  microvm,
+}: let
+  name = "lenovo-x1-carbon-gen11";
+  system = "x86_64-linux";
+  formatModule = nixos-generators.nixosModules.raw-efi;
+  generic-x86 = variant: extraModules: let
+    netvmExtraModules = [
+      {
+        microvm.devices = lib.mkForce [
+          {
+            bus = "pci";
+            path = "0000:00:14.3";
+          }
+        ];
+
+        # For WLAN firmwares
+        hardware.enableRedistributableFirmware = true;
+
+        networking.wireless = {
+          enable = true;
+
+          #networks."ssid".psk = "psk";
+        };
+      }
+    ];
+    guivmExtraModules = [
+      {
+        microvm.qemu.extraArgs = [
+          "-usb"
+          "-device"
+          "usb-host,vendorid=0x046d,productid=0xc52b"
+        ];
+        microvm.devices = [
+          {
+            bus = "pci";
+            path = "0000:00:02.0";
+          }
+        ];
+      }
+      ({pkgs, ...}: {
+        ghaf.graphics.weston.launchers = [
+          {
+            path = "${pkgs.waypipe}/bin/waypipe ssh -i ${pkgs.waypipe-ssh}/keys/waypipe-ssh -o StrictHostKeyChecking=no 192.168.101.5 chromium --enable-features=UseOzonePlatform --ozone-platform=wayland";
+            icon = "${pkgs.weston}/share/weston/icon_editor.png";
+          }
+
+          {
+            path = "${pkgs.waypipe}/bin/waypipe ssh -i ${pkgs.waypipe-ssh}/keys/waypipe-ssh -o StrictHostKeyChecking=no 192.168.101.6 gala --enable-features=UseOzonePlatform --ozone-platform=wayland";
+            icon = "${pkgs.weston}/share/weston/icon_editor.png";
+          }
+
+          {
+            path = "${pkgs.waypipe}/bin/waypipe ssh -i ${pkgs.waypipe-ssh}/keys/waypipe-ssh -o StrictHostKeyChecking=no 192.168.101.7 zathura";
+            icon = "${pkgs.weston}/share/weston/icon_editor.png";
+          }
+        ];
+      })
```

Comment:
can these live in the same definition as the modules/vms/... as discussed below? Can be a follow on patch

Reply:
SP-2840 is created to track this